src/rust/engine/run-all-tests.sh

```diff
@@ -7,21 +7,18 @@
 here=$(cd $(dirname "${BASH_SOURCE[0]}") && pwd)
 REPO_ROOT=$(dirname $(dirname $(dirname ${here})))
 
+source "${REPO_ROOT}/build-support/pants_venv"
 source "${REPO_ROOT}/build-support/bin/native/bootstrap.sh"
 
-source "${REPO_ROOT}/build-support/pants_venv"
 activate_pants_venv
-PANTS_SRCPATH="${REPO_ROOT}/src/python" prepare_to_build_native_code
-
-set +e
 
 exit_code=0
 
 for crate in $(find ${here} -name Cargo.toml); do
   (
     echo >&2 "Running tests for ${crate}:"
-    export RUST_BACKTRACE=1
-    "${CARGO_HOME}/bin/cargo" test --manifest-path=${crate}
+    RUST_BACKTRACE=1 PANTS_SRCPATH="${REPO_ROOT}/src/python" run_cargo test ${MODE_FLAG} \
```

Comment:
Did not mention this in the description, but specifying the same `MODE_FLAG` as `./pants` does indirectly through `bootstrap.sh` (ie: `--release` vs nothing, aka debug), was a part of this puzzle, even after which the `build.rs` rename _and_ removal of the `Cargo.toml ` `[project] build` were both necessary as well to achieve ~0s noop test times.